extensions/ui/org.eclipse.smarthome.ui.paper/src/main/java/org/eclipse/smarthome/ui/paper/internal/PaperUIApp.java

```diff
@@ -7,7 +7,14 @@
  */
 package org.eclipse.smarthome.ui.paper.internal;
 
+import org.eclipse.smarthome.io.net.http.HttpContextFactoryService;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
 import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Component;
```

Comment:
don't you need those as optional import in the manifest?

Reply:
yes. The import is also missing in the o.e.s.io.net MANIFEST. I dont know how that even worked.

Reply:
As long as they exist in the classpath on compile time no one will complain.
Missing annotation class definitions should not matter on runtime.

Reply:
anyhow: added and marked as optional.

Reply:
https://stackoverflow.com/a/3568041

Reply:
@maggu2810 thanks for the link. I was wondering about why the IDE did not complain. Although o.e.s.io.net was missing the import completely the service XML was generated and no compile error present.